src/Helpers/Group/GroupHelper.js

```diff
@@ -1,26 +1,42 @@
 
-import { getApprovalApi } from '../Shared/userLogin';
+import { getGroupApi } from '../Shared/userLogin';
 
-const userApi = getApprovalApi();
+const groupApi = getGroupApi();
 
 export async function fetchGroups() {
-  let groups = await userApi.fetchGroups();
+  let groupsData = await groupApi.listGroups();
+  let groups = groupsData.data;
+  console.log('Groups: ', groups);
   let len = groups.length;
   for (let idx = 0; idx < len; idx++) {
-    let users = await userApi.fetchUsersByGroupId(groups[idx].id);
-    groups[idx].members = users;
+    let groupWithUsers = await groupApi.getGroup(groups[idx].uuid);
+    groups[idx].members = groupWithUsers.principals;
   }
+
   return groups;
 }
 
 export async function updateGroup(data) {
-  await userApi.updateGroup(data.id, data);
+  await groupApi.updateGroup(data.uuid, data);
+  let members_list = data.members.map(user => { return user.username; });
+  //update the user members here - adding users and removing users from the group should be a separate action in the UI
+  let addUsers = data.user_list.filter(item => !members_list.includes(item.username));
+  let removeUsers = members_list.filter(item => !(data.user_list.map(user => { return user.username;})).includes(item));
+  if (addUsers.length > 0) {
+    await groupApi.addPrincipalToGroup(data.uuid, JSON.stringify({ principals: addUsers }));
+  }
+
+  if (removeUsers.length > 0) {
+    await groupApi.deletePrincipalFromGroup(data.uuid, removeUsers.join(','));
```

Comment:
instead of `join` i would use template string its a bit safer n case
```javascript
await groupApi.deletePrincipalFromGroup(data.uuid, `${removeUsers}`);
```

Reply:
I would keep the join for now, until the rbac service is more established with the principals management.